Analysis/Tasks/PWGDQ/v0selector.cxx

```diff
@@ -42,19 +44,275 @@ using namespace o2::framework;
+
+    //unit vector of (pep+pem)
+    float pl = dppair;
+    float ux = px / pl;
+    float uy = py / pl;
+    float uz = pz / pl;
+    float ax = uy / TMath::Sqrt(ux * ux + uy * uy);
+    float ay = -ux / TMath::Sqrt(ux * ux + uy * uy);
+
+    //momentum of e+ and e- in (ax,ay,az) axis. Note that az=0 by definition.
+    //Double_t ptep = iep->Px()*ax + iep->Py()*ay;
+    //Double_t ptem = iem->Px()*ax + iem->Py()*ay;
+
+    float pxep = px1;
+    float pyep = py1;
+    float pzep = pz1;
+    float pxem = px2;
+    float pyem = py2;
+    float pzem = pz2;
+
+    //vector product of pep X pem
+    float vpx = pyep * pzem - pzep * pyem;
+    float vpy = pzep * pxem - pxep * pzem;
+    float vpz = pxep * pyem - pyep * pxem;
+    float vp = TMath::Sqrt(vpx * vpx + vpy * vpy + vpz * vpz);
+
+    //unit vector of pep X pem
+    float vx = vpx / vp;
+    float vy = vpy / vp;
+    float vz = vpz / vp;
+
+    //The third axis defined by vector product (ux,uy,uz)X(vx,vy,vz)
+    float wx = uy * vz - uz * vy;
+    float wy = uz * vx - ux * vz;
+    // by construction, (wx,wy,wz) must be a unit vector.
+    // measure angle between (wx,wy,wz) and (ax,ay,0). The angle between them
+    // should be small if the pair is conversion
+    // this function then returns values close to pi!
+    float cosPhiV = wx * ax + wy * ay;
+    float phiv = TMath::ACos(cosPhiV);
+    return phiv;
+  }
+
+  float psipairv0(const array<float, 3>& ppos, const array<float, 3>& pneg, const float bz)
+  {
+    //Following idea to use opening of colinear pairs in magnetic field from e.g. PHENIX
+    //to ID conversions.
+    float mp[3] = {0, 0, 0};
+    float mn[3] = {0, 0, 0};
+
+    mp[0] = ppos[0];
+    mp[1] = ppos[1];
+    mp[2] = ppos[2];
+
+    mn[0] = pneg[0];
+    mn[1] = pneg[1];
+    mn[2] = pneg[2];
+
+    float deltat = TMath::ATan(mn[2] / (TMath::Sqrt(mn[0] * mn[0] + mn[1] * mn[1]) + 1.e-13)) - TMath::ATan(mp[2] / (TMath::Sqrt(mp[0] * mp[0] + mp[1] * mp[1]) + 1.e-13)); //difference of angles of the two daughter tracks with z-axis
+    float pEle = TMath::Sqrt(mn[0] * mn[0] + mn[1] * mn[1] + mn[2] * mn[2]);                                                                                                //absolute momentum val
+    float pPos = TMath::Sqrt(mp[0] * mp[0] + mp[1] * mp[1] + mp[2] * mp[2]);                                                                                                //absolute momentum val
+    float scalarproduct = mp[0] * mn[0] + mp[1] * mn[1] + mp[2] * mn[2];                                                                                                    //scalar product of propagated posit
+    float chipair = TMath::ACos(scalarproduct / (pEle * pPos));                                                                                                             //Angle between propagated daughter tracks
+    float psipair = TMath::Abs(TMath::ASin(deltat / chipair));
+    return psipair;
```

Comment:
You can combine these three lines and avoid assigning 2 unneeded variables

Reply:
If possoible, I prefer to keep chipair and psipair for visibility.